### PR TITLE
Style cleanup design doc/add user doc

### PR DESCRIPTION
**Pull request by jzulauf-lunarg**
Cleanup the the style of the design doc and add a user document suitable
for SDK documentation.

### CODE REVIEW

**Comment by christophe-lunarg** [comment]
synchronization validation is no listed in khronos_validation_layer.md.

Would it be possible to add it to that document and make that reference a link?

**Comment by christophe-lunarg** [comment]
I created a PR to solve this issue:https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/2044

File: docs/synchronization_usage.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+  </tr>
+</table>
+
+
+
+The specific areas covered by this layer are currently tracked in
+[Github Issue #72: Validate synchronization correctness: Sync Tracking](https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues/72)
+Requests for additional checks can be posted through the same issue, or by creating a new Github issue.
+
+## Enabling Synchronization Validation
```
Comment (KarenGhavam-lunarG) [inline docs/synchronization_usage.md:73]: I think the preference is to emphasize using Vulkan Configurator for enabling features of validation. And then to document how to do it with environment variables, programmatically, etc. However we can make those changes as "round two"

File: docs/synchronization_usage.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+VkValidationFeaturesEXT features = {};
+features.sType = VK_STRUCTURE_TYPE_VALIDATION_FEATURES_EXT;
+features.enabledValidationFeatureCount = 1;
+features.pEnabledValidationFeatures = enables;
+
+VkInstanceCreateInfo info = {};
+info.pNext = &features;
+```
+
+## Synchronization Validation Functionality
```
Comment (KarenGhavam-lunarG) [inline docs/synchronization_usage.md:110]: It feels like this section should be higher in the document...

File: docs/synchronization_usage.md (hunk 1 above)
Comment (jzulauf-lunarg) [reply docs/synchronization_usage.md:96]: This is a direct copy of the Best Practices content, touched up to point to Sync Val controls

File: docs/synchronization_usage.md (hunk 2 above)
Comment (jzulauf-lunarg) [reply docs/synchronization_usage.md]: moved

**Comment by jzulauf-lunarg** [comment]
@KarenGhavam-lunarG  -- I think I've integrate your feedback.

**Comment by jzulauf-lunarg** [comment]
@KarenGhavam-lunarG -- merged

**Pull request merged**